### PR TITLE
[GPU] Pass lrn unit tests on DG2

### PR DESCRIPTION
**Pull request by tuxedcat**
### Details:
 - Change Scale to Eltwise(mode::prod) in unit test topologies.
 - Change ScaleWithBias to Eltwise(mode::prod)+Eltwise(mode::sum) in unit test topologies.
 - Support Gather-Eltwise fusing.

### Tickets:
 - 67491

### CODE REVIEW

**Comment by sungeunk** [comment]
It looks good to me.

File: src/plugins/intel_gpu/src/graph/activation.cpp [hunk 1]
```diff
@@ -30,7 +30,8 @@ layout activation_inst::calc_output_layout(activation_node const& node) {
         activation_func::floor,
         activation_func::clamp };
 
-    if (input_node_layout.data_type == data_types::i8 || input_node_layout.data_type == data_types::i32) {
+    if (input_node_layout.data_type == data_types::i8 || input_node_layout.data_type == data_types::u8 ||
+        input_node_layout.data_type == data_types::i32) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/activation.cpp:34]: Could you explain why this change is necessary?

Comment (tuxedcat) [reply src/plugins/intel_gpu/src/graph/activation.cpp:34]: Kernel compile fails when activation=exp and data_type=u8. I thought restriction for i8 should be applied to u8 too.

Comment (isanghao) [reply src/plugins/intel_gpu/src/graph/activation.cpp:34]: Then, what happens with this change? Does it fallback to different kernel?

Comment (isanghao) [reply src/plugins/intel_gpu/src/graph/activation.cpp:34]: Discussed offline: 
This change is added to show better message in case of failure. Previously, error message was from kernel compilation. Now error message is from here. No behavioral change is expected.

**Review by isanghao — APPROVED** [review]
